File: TIA/tia_final_project/src/repositories/population/PopulationCopy.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package repositories.population;

/**
 *
 * @author crdzbird
 */

import java.util.List;
import repositories.IndividualImplementation;
import repositories.elitism.ElitismInterface;

public class PopulationCopy extends PopulationImplementation<IndividualImplementation> {

    public PopulationCopy(List<IndividualImplementation> initpop, ElitismInterface<IndividualImplementation> elitism) {
        super(initpop, elitism);
    }

    @Override
    public PopulationInterface<IndividualImplementation> copy() {
        return new PopulationCopy(this.population, this.elitism);
    }
}

File: TIA/tia_genetic_memetic/lib/repositories/population/population_copy.dart
import 'package:tia_genetic_memetic/repositories/elitism/elitism_interface.dart';
import 'package:tia_genetic_memetic/repositories/individual/individual_implementation.dart';
import 'package:tia_genetic_memetic/repositories/population/population_implementation.dart';
import 'package:tia_genetic_memetic/repositories/population/population_interface.dart';

class PopulationCopy
    extends PopulationImplementation<IndividualImplementation> {
  PopulationCopy(List<IndividualImplementation> initpop,
      ElitismInterface<IndividualImplementation> elitism)
      : super(initpop, elitism);

  @override
  PopulationInterface<IndividualImplementation> copy() {
    return PopulationCopy(
        List<IndividualImplementation>.from(population), elitism);
  }
}
